### PR TITLE
Enable email account linking

### PR DESCRIPTION
**Pull request by nicklathe**
This PR wires up the link email form with the account linking controller. It adds the following

- Pass email as param to form
- Refactor form to post to linking route
- Fix areas where variables weren't being passed correctly
- Remove un-used hidden form field with hashed email
- Fix unit test to include email in url param

## Links

[Jira task](https://codedotorg.atlassian.net/browse/P20-960)

## Testing story

```
yarn test:unit --entry=./test/unit/lib/ui/lti/link/LtiLinkAccountPage/LtiLinkAccountPageTest.tsx

Finished in 3.911 secs / 1.918 secs @ 16:41:40 GMT+0000 (Coordinated Universal Time)

SUMMARY:
✔ 3 tests completed

Done.
```

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/app/views/lti/v1/account_linking/landing.haml [hunk 1]
```diff
@@ -1,8 +1,8 @@
 - content_for(:head) do
   :ruby
     script_data = {}
-    script_data[:lti_provider] = params[:lti_provider]
-    script_data[:lti_provider_name] = Policies::Lti::LMS_PLATFORMS[params[:lti_provider].to_sym][:name]
+    script_data[:lti_provider] = lti_provider
```
Comment (carl-codeorg) [inline dashboard/app/views/lti/v1/account_linking/landing.haml:4]: Nice, I had to make these same changes in my PR to get these params to pass properly. Good to see it wasn't just my machine

**Review by carl-codeorg — APPROVED** [review]
LGTM. Pre-populating the email was a great idea; forgot we had that data point

File: dashboard/app/views/lti/v1/account_linking/landing.haml (hunk 1 above)
Comment (nicklathe) [reply dashboard/app/views/lti/v1/account_linking/landing.haml:4]: Yeah, took me a minute and had to learn what this line was doing (setting the locals values) - https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/controllers/lti_v1_controller.rb#L202

Comment (nicklathe) [reply dashboard/app/views/lti/v1/account_linking/landing.haml:4]: My assumption is that's the correct way to pass variables to the view in a redirect, where setting an instance variable probably isn't accessible in the view due to the redirect? 🤷‍♂️

**Pull request merged**